utils/src/big_bit_ops.rs

```diff
@@ -0,0 +1,114 @@
+//! This module provides a set of functions to perform bit operations on big integers.
```

Comment:
Suggest creating `biguint_helpers.rs` containing all of this implemented as a `BigUintHelpers` trait instead